android_15/src/org/ros/android/view/visualization/layer/PathLayer.java

```diff
@@ -79,27 +81,21 @@ public void onNewMessage(nav_msgs.Path path) {
 
   private void updateVertexBuffer(nav_msgs.Path path) {
     ByteBuffer goalVertexByteBuffer =
-        ByteBuffer.allocateDirect(path.getPoses().size() * 3 * Float.SIZE / 8);
```

Comment:
Right. I think that logic was wrong, probably due to the lack of using a constant for the separation between points.